scalanet/src/io/iohk/scalanet/peergroup/dynamictls/DynamicTLSPeerGroupInternals.scala

```diff
@@ -196,10 +188,9 @@ private[dynamictls] object DynamicTLSPeerGroupInternals {
     override def close(): Task[Unit] = {
       for {
         _ <- Task.now(log.debug("Closing client channel to peer {}", peerInfo))
-        _ <- Task(messageSubject.onComplete())
         ctx <- Task.fromFuture(activationF)
         _ <- toTask(ctx.close())
-        _ <- Task.fromFuture(deactivationF)
+        _ <- toTask(ctx.closeFuture())
```

Comment:
Out of lack of knowledge, why are both close needed here?

Reply:
we we want to be sure that `channelnactive` handler had run before returning from task. In theory it should happen just with `toTask(ctx.close())` but to be sure we are are also waiting on `closeFuture` which is actually and promise.